### PR TITLE
[INLONG-380][Sort] Add lightweight sort and transform related instructions

### LINKED ISSUE #380: [Sort] Optimize sort installation and deployment documentation
#### Expected behavior

Tell us what should happen

#### Actual behavior

Tell us what happens instead

#### Steps to reproduce

How can we reproduce the issue

#### System configuration
**InLong version**: x.y

### PR DESCRIPTION
**Pull request by yunqingmoswu**
Fixes #380 

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve.*

### Modifications

*Describe the modifications you've done.*

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

File: docs/modules/sort/overview.md [hunk 1]
```diff
@@ -9,24 +9,34 @@ Inlong-sort is simply an Flink application, and relys on Inlong-manager to manag
...
 
+## supported transforms
+- String split index
+- String regular replace 
+- String regular replace the first value
+- Data filter
+- Data distinct
+- Regular join
+
 ## supported sources
```
Comment (dockerzhang) [inline docs/modules/sort/overview.md:20]: sources -> Extract Node

Comment (dockerzhang) [inline docs/modules/sort/overview.md:20]: sources -> Extract Node

File: docs/modules/sort/overview.md [hunk 2]
```diff
@@ -9,24 +9,34 @@ Inlong-sort is simply an Flink application, and relys on Inlong-manager to manag
...
-- pulsar
+- Pulsar
+- MySQL
+- Kafka
 
 ## supported storages
```
Comment (dockerzhang) [inline docs/modules/sort/overview.md:25]: storages
->
Load Node

Comment (dockerzhang) [inline docs/modules/sort/overview.md:25]: storages
->
Load Node

File: docs/modules/sort/overview.md [hunk 3]
```diff
@@ -9,24 +9,34 @@ Inlong-sort is simply an Flink application, and relys on Inlong-manager to manag
...
+- Hive
+- Kafka
+- ClickHouse
+- Iceberg
 
 ## limitations
 Currently, we just support extracting specified fields in the stage of **Transform**.
 
 ## future plans
 ### More kinds of source systems
```
Comment (dockerzhang) [inline docs/modules/sort/overview.md:35]: source systems -> Extract Node

Comment (dockerzhang) [inline docs/modules/sort/overview.md:35]: source systems -> Extract Node

File: docs/modules/sort/overview.md [hunk 4]
```diff
@@ -9,24 +9,34 @@ Inlong-sort is simply an Flink application, and relys on Inlong-manager to manag
...
-kafka and etc
+Oracle,  Mongo DB, SqlServer, and etc
 
 ### More kinds of storage systems
```
Comment (dockerzhang) [inline docs/modules/sort/overview.md:38]: storage systems
->
Load Node

Comment (dockerzhang) [inline docs/modules/sort/overview.md:38]: storage systems
->
Load Node

File: docs/modules/sort/quick_start.md [hunk 5]
```diff
@@ -20,9 +20,7 @@ Now you can submit job to flink with the jar compiled, refer to [how to submit j
 Example：
 ```
 ./bin/flink run -c org.apache.inlong.sort.singletenant.flink.Entrance inlong-sort/sort-[version].jar \
```
Comment (dockerzhang) [inline docs/modules/sort/quick_start.md:22]: org.apache.inlong.sort.singletenant.flink.Entrance
->
org.apache.inlong.sort.flink.Entrance

singletenant have no meaning start from this version now, we should rename the entrance class name.

Comment (dockerzhang) [inline docs/modules/sort/quick_start.md:22]: org.apache.inlong.sort.singletenant.flink.Entrance
->
org.apache.inlong.sort.flink.Entrance

singletenant have no meaning start from this version now, we should rename the entrance class name.

File: docs/modules/sort/quick_start.md [hunk 6]
```diff
@@ -32,35 +30,17 @@ Notice：
 - `inlong-sort/sort-[version].jar` is the compiled jar
 
 ## Necessary configurations
-- `--cluster-id ` represent a specified inlong-sort application, same as the configuration of `sort.appName` in inlong-manager
-- `--dataflow.info.file` dataflow configuration file path
-- `--source.type` source of the application, currently "pulsar" is supported
-- `--sink.type` sink of the application, currently "clickhouse", "hive", "iceberg", "kafka" are supported
-- `--metrics.audit.proxy.hosts` audit proxy host address for reporting audit metrics
+- `--group.info.file` dataflow configuration file path
+- `--lightweight` whether to use lightweight sort,default false
```
Comment (dockerzhang) [inline docs/modules/sort/quick_start.md:34]: we will use the lightweight version all the time, `--lightweight` is useless?

Comment (dockerzhang) [inline docs/modules/sort/quick_start.md:34]: we will use the lightweight version all the time, `--lightweight` is useless?

File: docs/modules/sort/quick_start.md [hunk 7]
```diff
@@ -32,35 +30,17 @@ Notice：
...
 
 **Example**
 ```
---cluster-id debezium2kafka-canal --dataflow.info.file /YOUR_DATAFLOW_INFO_DIR/debezium-to-kafka-canal.json \
---source.type pulsar --sink.type kafka
+--lightweight true --group.info.file /YOUR_DATAFLOW_INFO_DIR/mysql-to-kafka.json
```
Comment (dockerzhang) [inline docs/modules/sort/quick_start.md:38]: ditto for `--lightweight`

Comment (dockerzhang) [inline docs/modules/sort/quick_start.md:38]: ditto for `--lightweight`

File: docs/modules/sort/overview.md [hunk 8]
```diff
@@ -9,24 +9,34 @@ Inlong-sort is simply an Flink application, and relys on Inlong-manager to manag
 
 # features
 
+## supported transforms
+- String split index
```
Comment (dockerzhang) [inline docs/modules/sort/overview.md:13]: it's better to add an example to describe the transform operation.

File: docs/modules/sort/overview.md [hunk 9]
```diff
@@ -9,24 +9,34 @@ Inlong-sort is simply an Flink application, and relys on Inlong-manager to manag
...
-- hive（Currently we support parquet, orc and text file format）
-- kafka
-- clickhouse
-- iceberg
+- Hive
+- Kafka
+- ClickHouse
+- Iceberg
 
 ## limitations
```
Comment (dockerzhang) [inline docs/modules/sort/overview.md:31]: do we have this limitation for the lightweight sort now?

File: docs/modules/sort/dataflow_example.md [hunk 10]
```diff
@@ -7,305 +7,1717 @@ sidebar_position: 3
 
 To make it easier for you to create inlong-sort jobs, here we list some dataflow configuration examples.
 
-## Pulsar to Kafka
+## MySQL to Kafka
 
-Normal example:
+- Normal example:
```
Comment (dockerzhang) [inline docs/modules/sort/dataflow_example.md:12]: what `Normal` meaning is here? is it needless?

File: docs/modules/sort/dataflow_example.md [hunk 11]
```diff
@@ -7,305 +7,1717 @@ sidebar_position: 3
...
+      ]
     }
+  ]
 }
 ```
 
-Whole-database migration example：
+## Kafka to Hive
+
+- Normal example:
```
Comment (dockerzhang) [inline docs/modules/sort/dataflow_example.md:274]: `- Normal example:` should remove

File: docs/modules/sort/overview.md (hunk 1 above)
Comment (yunqingmoswu) [reply docs/modules/sort/overview.md:20]: It is a good idea,I will do it latter.

File: docs/modules/sort/overview.md (hunk 2 above)
Comment (yunqingmoswu) [reply docs/modules/sort/overview.md:25]: It is a good idea,I will do it latter.

File: docs/modules/sort/overview.md (hunk 1 above)
Comment (yunqingmoswu) [reply docs/modules/sort/overview.md]: done

File: docs/modules/sort/overview.md (hunk 2 above)
Comment (yunqingmoswu) [reply docs/modules/sort/overview.md]: done

File: docs/modules/sort/overview.md (hunk 3 above)
Comment (yunqingmoswu) [reply docs/modules/sort/overview.md]: done

Comment (yunqingmoswu) [reply docs/modules/sort/overview.md]: done

File: docs/modules/sort/quick_start.md (hunk 5 above)
Comment (yunqingmoswu) [reply docs/modules/sort/quick_start.md]: done

File: docs/modules/sort/quick_start.md (hunk 6 above)
Comment (yunqingmoswu) [reply docs/modules/sort/quick_start.md]: Yes,I will do it latter

Comment (yunqingmoswu) [reply docs/modules/sort/quick_start.md]: done

File: docs/modules/sort/quick_start.md (hunk 7 above)
Comment (yunqingmoswu) [reply docs/modules/sort/quick_start.md]: done

File: docs/modules/sort/overview.md (hunk 9 above)
Comment (yunqingmoswu) [reply docs/modules/sort/overview.md]: We do not have this limitation, i remove it latter

File: docs/modules/sort/dataflow_example.md (hunk 10 above)
Comment (yunqingmoswu) [reply docs/modules/sort/dataflow_example.md]: It is needless,i remove it

File: docs/modules/sort/overview.md (hunk 8 above)
Comment (yunqingmoswu) [reply docs/modules/sort/overview.md]: The demo of Transform is presented in dataflow_example.md

File: i18n/zh-CN/docusaurus-plugin-content-docs/current/modules/sort/overview.md [hunk 12]
```diff
@@ -4,31 +4,43 @@ sidebar_position: 1
...
+### 支持更多种类的 Extract Node
+Oracle, SqlServer等。
 
-### 支持更多种类的存储
-Hbase，Elastic Search等
+### 支持更多种类的 Transform 
+时间窗口聚合, 内容提取, 类型转换, 时间格式转换等。
 
+### 支持更多种类的 Load Node 
+Elastic Search等。
```
Comment (dockerzhang) [inline i18n/zh-CN/docusaurus-plugin-content-docs/current/modules/sort/overview.md:45]: Elastic Search
->
Elasticsearch

**Review by dockerzhang — APPROVED** [review]

File: i18n/zh-CN/docusaurus-plugin-content-docs/current/modules/sort/overview.md [hunk 13]
```diff
@@ -4,31 +4,43 @@ sidebar_position: 1
...
+- Iceberg
+- PostgreSQL
+- FileSystem
 
 ## 未来规划
-### 支持更多种类的数据源
-kafka等
-
+### 支持更多种类的 Extract Node
+Oracle, SqlServer等。
```
[the remark this answers is not in the document]
Comment (yunqingmoswu) [reply i18n/zh-CN/docusaurus-plugin-content-docs/current/modules/sort/overview.md]: done

File: i18n/zh-CN/docusaurus-plugin-content-docs/current/modules/sort/overview.md (hunk 12 above)
Comment (yunqingmoswu) [reply i18n/zh-CN/docusaurus-plugin-content-docs/current/modules/sort/overview.md]: done

File: docs/modules/sort/overview.md [hunk 14]
```diff
@@ -3,30 +3,43 @@ title: Overview
...
-
-### More kinds of storage systems
-Hbase, Elastic Search, and etc
-
-### More kinds of file format in hive sink
-sequence file
\ No newline at end of file
+# Overview
+InLong-Sort is used to extract data from different source systems, then transforms the data and finally loads the data into diffrent storage systems.
+InLong-Sort is simply an Flink application, and relys on InLong-Manager to manage meta data(such as the source informations and storage informations).
```
Comment (EMsnap) [inline docs/modules/sort/overview.md:8]: typo an -> a

Comment (yunqingmoswu) [reply docs/modules/sort/overview.md]: done

File: docs/modules/sort/dataflow_example.md [hunk 15]
```diff
@@ -5,307 +5,1717 @@ sidebar_position: 3
...
+                }
+            ]
+        }
+    ]
+}
+```
+
+## Transform examples
+
+Currently only supports string split, string regex replace, string regex replace first matched value, data distinct,data filter, regular join, and etc.
```
Comment (thexiay) [inline docs/modules/sort/dataflow_example.md:428]: add a link to support transformation:
https://github.com/yunqingmoswu/incubator-inlong-website/blob/INLONG-380/docs/modules/sort/overview.md#supported-transform

File: docs/modules/sort/quick_start.md [hunk 16]
```diff
@@ -3,64 +3,43 @@ title: Deployment
...
 
 ## Necessary configurations
-- `--cluster-id ` represent a specified inlong-sort application, same as the configuration of `sort.appName` in inlong-manager
-- `--dataflow.info.file` dataflow configuration file path
-- `--source.type` source of the application, currently "pulsar" is supported
-- `--sink.type` sink of the application, currently "clickhouse", "hive", "iceberg", "kafka" are supported
-- `--metrics.audit.proxy.hosts` audit proxy host address for reporting audit metrics
-
+- `--group.info.file` dataflow configuration file path
+-
```
Comment (thexiay) [inline docs/modules/sort/quick_start.md:34]: Redundant `-`

File: docs/modules/sort/quick_start.md [hunk 17]
```diff
@@ -3,64 +3,43 @@ title: Deployment
...
 - `inlong-sort/sort-[version].jar` is the compiled jar
 
 ## Necessary configurations
-- `--cluster-id ` represent a specified inlong-sort application, same as the configuration of `sort.appName` in inlong-manager
-- `--dataflow.info.file` dataflow configuration file path
-- `--source.type` source of the application, currently "pulsar" is supported
-- `--sink.type` sink of the application, currently "clickhouse", "hive", "iceberg", "kafka" are supported
-- `--metrics.audit.proxy.hosts` audit proxy host address for reporting audit metrics
-
+- `--group.info.file` dataflow configuration file path
```
Comment (thexiay) [inline docs/modules/sort/quick_start.md:33]: notice if we need out Extended extrace and load，we should put our connector into ${FLINK_HOME}/lib folder.For example, if we want mysql to hive，we shoule put `sort-connector-mysql-cdc.jar` and `sort-connector-hive.jar` in ${FLINK_HOME}/lib

**Review by EMsnap — APPROVED** [review]

File: docs/modules/sort/quick_start.md (hunk 16 above)
Comment (yunqingmoswu) [reply docs/modules/sort/quick_start.md]: done

File: docs/modules/sort/quick_start.md (hunk 17 above)
Comment (yunqingmoswu) [reply docs/modules/sort/quick_start.md:33]: It is not necessary, these dependencies will be automatically imported after the sort-dist-[version].jar package is introduced

File: docs/modules/sort/dataflow_example.md (hunk 15 above)
Comment (yunqingmoswu) [reply docs/modules/sort/dataflow_example.md:428]: It is a good idea

**Review by thexiay — APPROVED** [review]

**Pull request merged**